### ISSUE TITLE
coq_makefile doesn't pass -R options to coqdoc, breaks links

### ISSUE THREAD

**Issue by yforster** [coq/coq#6697]
#### Version

Coq 8.7.1

#### Operating system

Linux 4.14.15-1-ARCH x86_64

#### Description of the problem

Hi all,

when using a _CoqProject file which includes COQDOCFLAGS = "...", as does for instance https://github.com/tebbi/coqdocjs/, links in the resulting html are broken (i.e. <a> fields are not generated).

Example _CoqProject:

```
-R . Top

COQDOCFLAGS = ""

test.v
```
Example file test.v
```
Lemma test : forall n : nat, 0 + n = n.
Proof.
  eapply plus_O_n.
Qed.
```

Now `coq_makefile -f _CoqProject -o Makefile && make html` generates a `html/test.html` file where the link to `plus_O_n` is broken. This can be fixed by adding `-R . Top` to the COQDOCFLAGS, which seems unnecessarily redundant, and wasn't necessary with the old `coq_makefile` (I think, at least it wasn't necessary under 8.6).

Is that intended behaviour?

Thanks in advance!
Yannick

**Comment by emiquey** [comment]
I have been in trouble with coqdoc for the same reason, thanks for pointing this issue.

**Comment by Zimmi48** [comment]
@gares Do you think you'll have time to look at this issue (and hopefully fix it) for 8.7.2 or should we tag it "help wanted"?

**Comment by yforster** [comment]
I looked into this a bit further, maybe that helps Enrico. My impression is that replacing
```
COQDOCFLAGS?=-interpolate -utf8 $(COQLIBS_NOML)
```
in line 172 of `tools/CoqMakefile.in` with
```
COQDOCFLAGS+=$(COQLIBS_NOML)
COQDOCFLAGS?=-interpolate -utf8
```
fixes the problem. But the html target still mentions `$(COQDOCLIBS)`, and I can't find any other mention of this anywhere else, so maybe that's the culprit and it should really be `$(COQLIBS_NOML)` there as well.

**Comment by Zimmi48** [comment]
Cool! Then I suggest you open a pull request with this fix and we take it from there.

**Comment by RalfJung** [comment]
This is essentially a duplicate of #6659

**Comment by Zimmi48** [comment]
Yes, except that this expressed that this is not an enhancement-request but rather a regression. So the fix is eligible to be backported in v8.7.
